### PR TITLE
Add C implementation for `@stdlib/math/base/assert/is-even`

### LINKED ISSUE #1666: [RFC]: Add C implementation for `@stdlib/math/base/assert/is-even`
### Description

This RFC proposes adding native C implementation for [`@stdlib/math/base/assert/is-even`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/assert/is-even/).

```c
double stdlib_base_is_even( const double x )
```

### Related Issues

None

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by aman-095**
Resolves #1666.

## Description

> What is the purpose of this pull request?

This pull request adds native C implementation for
`@stdlib/math/base/assert/is-even`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1666 
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/assert/is-even/README.md [hunk 1]
```diff
@@ -104,6 +104,97 @@ for ( i = 0; i < 100; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/assert/is_even.h"
+```
+
+#### stdlib_base_is_even( x )
+
+Test if a finite numeric value is an even number.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/README.md]:
```suggestion
Tests if a finite numeric value is an even number.
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/c/native/Makefile [hunk 2]
```diff
@@ -0,0 +1,146 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/c/native/Makefile]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/c/native/benchmark.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+double benchmark() {
+	double elapsed;
+	double x;
+	double t;
+	bool b;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( rand_double() * 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/benchmark/c/native/benchmark.c]:
```suggestion
		x = ( rand_double() * 200.0 ) - 100.0;
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/examples/c/example.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+int main( void ) {
+    double x;
+    bool v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 );
+        v = stdlib_base_is_even( x );
+        printf( "x = %lf, is_even(x) = %s\n", x, ( v ) ? "true" : "false" );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/examples/c/example.c]:
```suggestion
        printf( "x = %lf, is_even(x) = %s\n", x, ( v ) ? "even" : "not even" );
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/include/stdlib/math/base/assert/is_even.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Test if a finite numeric value is an even number.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/include/stdlib/math/base/assert/is_even.h]:
```suggestion
* Tests if a finite numeric value is an even number.
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/lib/native.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+// MODULES //
+
+var Boolean = require( '@stdlib/boolean/ctor' );
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Test if a finite numeric value is an even number.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/lib/native.js]:
```suggestion
* Tests if a finite numeric value is an even number.
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/src/main.c [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+* @return	  output value
+*
+* @example
+* #include <stdbool.h>
+*
+* bool out = stdlib_base_is_even( 3.0 );
+* // returns false
+*/
+bool stdlib_base_is_even( const double x ) {
+    return ( stdlib_base_is_integer ( x/2.0 ) );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/src/main.c]:
```suggestion
    return stdlib_base_is_integer( x/2.0 );
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/test/test.native.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var randu = require( '@stdlib/random/base/randu' );
+var round = require( '@stdlib/math/base/special/round' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/test/test.native.js]:
```suggestion
var randu = require( '@stdlib/random/base/randu' );
var round = require( '@stdlib/math/base/special/round' );
var PINF = require( '@stdlib/constants/float64/pinf' );
var NINF = require( '@stdlib/constants/float64/ninf' );
```

File: lib/node_modules/@stdlib/math/base/assert/is-even/test/test.native.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof isEven, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `true` if provided an even number', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-even/test/test.native.js]:
```suggestion
tape( 'the function returns `true` if provided an even number', opts, function test( t ) {
```

**Review by Pranavchiku — APPROVED** [review]
Thanks @aman-095 for this! This is ready to merge.

**Comment by Planeshifter** [comment]
@aman-095 Thanks much for contributing so many C implementations already, it's much appreciated! 🙌 

Two small nits for future reference:

1) It's not necessary to update the copyright year in an existing file like the `README.md` when adding new content to it; we don't update them usually to keep track on when the files were originally added.
2) Would you mind creating PRs such as this with a title such as this:

> feat: add C implementation for `math/base/assert/is-even`

Then the suggested commit message when squashing and merging will be auto-populated with the desired format per our [git commit message style guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/style-guides/git/README.md) already, making it easier to merge them.

Thanks again!

**Pull request merged**